README.md

```diff
@@ -1,7 +1,145 @@
-# MRL
+# [Matryoshka Representations for Adaptive Deployment](https://arxiv.org/abs/2205.13147)
+Aditya Kusupati, Gantavya Bhatt, Aniket Rege, Matthew Wallingford, Aditya Sinha, Vivek Ramanujan, William Howard-Snyder, Kaifeng Chen, Sham Kakade, Prateek Jain, Ali Farhadi
 
-This project is primarily organized into 
-1. Training MRL models
-2. Evaluating MRL models via image retrieval, few shot learning
-3. Analysis of MRL models via superclass hierarchy and oracle performance
+This repository contains the code for training and evaluating Matryoshka Representation with CNN backbone, along with visualizations. Training code is built upon [FFCV](https://github.com/libffcv/ffcv-imagenet) modified for MRL. The repository is organized as follows:
 
+1. Set up
+2. Matryoshka Linear Layer
+3. Training ResNet50 Models
+4. Evaluating Trained Models
+5. Evaluating pre-trained checkpoints
+5. Retrieval Performance
+
+Lastly, we also provide code to generate the figures mentioned in our paper. Please have a look at thee `Visualization/` folder. 
+## Set Up
+First pip install the requirements file in this directory:
+```
+pip install -r requirements.txt
+```
+We use PyTorch Distributed Data Parallel shared over 2 A100 GPUs and FFCV for the dataloader. Please make sure to scale the learning rate according to the #GPUs used. 
```

Comment:
We can write that we linearly downscaled the original learning rate by 4 as we had 4x less GPU than original FFCV benchmark

Reply:
Incorporating.